pkg/acl/acl.go

```diff
@@ -0,0 +1,422 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//+build linux
+
+// Package acl is a wrapper over libacl that dlopens it instead of being linked
+// to it. The code is based on go-acl by Joseph Naegele
+// (https://github.com/naegelejd/go-acl).
+package acl
+
+// #cgo LDFLAGS: -ldl
+// #include <stdlib.h>
+// #include <dlfcn.h>
+// #include <stdio.h>
+// #include <sys/acl.h>
+// #include <sys/types.h>
+// #include <unistd.h>
+//
+// acl_t
+// my_acl_from_text(void *f, const char *acl)
+// {
+//   acl_t (*acl_from_text)(const char *buf_p);
+//
+//   acl_from_text = (acl_t (*)(const char *))f;
+//   return acl_from_text(acl);
+// }
+//
+// int
+// my_acl_set_file(void *f, const char *path_p, acl_type_t type, acl_t acl)
+// {
+//   int (*acl_set_file)(const char *path_p, acl_type_t type, acl_t acl);
+//
+//   acl_set_file = (int (*)(const char *, acl_type_t, acl_t))f;
+//   return acl_set_file(path_p, type, acl);
+// }
+//
+// int
+// my_acl_free(void *f, acl_t acl)
+// {
+//   int (*acl_free)(acl_t acl);
+//
+//   acl_free = (int (*)(acl_t))f;
+//   return acl_free(acl);
+// }
+//
+// int
+// my_acl_valid(void *f, acl_t acl)
+// {
+//   int (*acl_valid)(acl_t acl);
+//
+//   acl_valid = (int (*)(acl_t))f;
+//   return acl_valid(acl);
+// }
+//
+// int
+// my_acl_create_entry(void *f, acl_t *acl_p, acl_entry_t *entry_p)
+// {
+//   int (*acl_create_entry)(acl_t *acl_p, acl_entry_t *entry_p);
+//
+//   acl_create_entry = (int (*)(acl_t *, acl_entry_t *))f;
+//   return acl_create_entry(acl_p, entry_p);
+// }
+//
+// int
+// my_acl_set_tag_type(void *f, acl_entry_t entry_d, acl_tag_t tag_type)
+// {
+//   int (*acl_set_tag_type)(acl_entry_t entry_d, acl_tag_t tag_type);
+//
+//   acl_set_tag_type = (int (*)(acl_entry_t, acl_tag_t))f;
+//   return acl_set_tag_type(entry_d, tag_type);
+// }
+//
+// int
+// my_acl_get_permset(void *f, acl_entry_t entry_d, acl_permset_t *permset_p)
+// {
+//   int (*acl_get_permset)(acl_entry_t entry_d, acl_permset_t *permset_p);
+//
+//   acl_get_permset = (int (*)(acl_entry_t, acl_permset_t *))f;
+//   return acl_get_permset(entry_d, permset_p);
+// }
+//
+// int
+// my_acl_add_perm(void *f, acl_permset_t permset_d, acl_perm_t perm)
+// {
+//   int (*acl_add_perm)(acl_permset_t permset_d, acl_perm_t perm);
+//
+//   acl_add_perm = (int (*)(acl_permset_t, acl_perm_t))f;
+//   return acl_add_perm(permset_d, perm);
+// }
+import "C"
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"unsafe"
+)
+
+const (
+	otherExec  = 1 << iota
+	otherWrite = 1 << iota
+	otherRead  = 1 << iota
+	groupExec  = 1 << iota
+	groupWrite = 1 << iota
+	groupRead  = 1 << iota
+	userExec   = 1 << iota
+	userWrite  = 1 << iota
+	userRead   = 1 << iota
+
+	TagUserObj  Tag = C.ACL_USER_OBJ
```

Comment:
Consider putting these values in a separate `const` block. Then you don't have to repeat that `Tag` word after the first member.